### PR TITLE
cargo-show-asm: unbreak on darwin

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

also adds me as a maintainer

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rrbutani** [comment]
Result of `nixpkgs-review pr 196861` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-show-asm</li>
  </ul>
</details>

**Comment by rrbutani** [comment]
Result of `nixpkgs-review pr 196861` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-show-asm</li>
  </ul>
</details>

File: pkgs/development/tools/rust/cargo-show-asm/default.nix [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-, fetchFromGitHub
 , rustPlatform
-, pkg-config
+, fetchFromGitHub
+, curl
 , installShellFiles
+, pkg-config
 , openssl
+, stdenv
+, darwin
```
Comment (oxalica) [inline pkgs/development/tools/rust/cargo-show-asm/default.nix:9]: I'm against reordering the existing one before a tree-wide enforced formatting spec.

File: pkgs/development/tools/rust/cargo-show-asm/default.nix [hunk 2]
```diff
@@ -21,14 +24,25 @@ rustPlatform.buildRustPackage rec {
 
   cargoHash = "sha256-S7OpHNjiTfQg7aPmHEx6Q/OV5QA9pB29F3MTIeiLAXg=";
 
-  nativeBuildInputs = [ pkg-config installShellFiles ];
-  buildInputs = [ openssl ];
+  nativeBuildInputs = [
+    curl.dev
```
Comment (oxalica) [inline pkgs/development/tools/rust/cargo-show-asm/default.nix:28]: Is both `curl.dev` and `curl` below required? Using it in both `nativeBuildInputs` and `buildInputs` is kind of weird.

**Review by oxalica — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/rust/cargo-show-asm/default.nix (hunk 2 above)
Comment (figsoda) [reply pkgs/development/tools/rust/cargo-show-asm/default.nix:28]: curl-sys calls curl-config to check for features and falls back to vendorred version when it fails

File: pkgs/development/tools/rust/cargo-show-asm/default.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/development/tools/rust/cargo-show-asm/default.nix:9]: even though not strictly enforced, there is a sorting convention for nixpkgs

> Arguments should be listed in the order they are used, with the exception of lib, which always goes first.

https://nixos.org/manual/nixpkgs/unstable/#sec-syntax

**Review by Br1ght0ne — APPROVED** [review]
Result of `nixpkgs-review pr 196861` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cargo-show-asm</li>
  </ul>
</details>

---

LGTM, works fine on my M1.

**Pull request merged**